repair/repair.hh

```diff
@@ -226,6 +226,8 @@ private:
     std::atomic_bool _shutdown alignas(seastar::cache_line_size);
     // Map repair id into repair_info.
     std::unordered_map<int, lw_shared_ptr<repair_info>> _repairs;
+    std::unordered_set<utils::UUID> _pending_repairs;
+    std::unordered_set<utils::UUID> _aborted_pending_repairs;
```

Comment:
I suggest to change _repairs to std::unordered_map<utils::UUID, lw_shared_ptr<repair_info>>. Whether it's aborted is a bool in repair_info. Then things like is_aborted() become part of repair_info.

Reply:
The problem is that we create repair_info late in the process. When one request abort, the repair_info is not create yet.

Reply:
Why not create it early? it should be the central point of state, rather than distributing it over many places.

Reply:
Repair job associated with a user request (with a repair id) is per node. In a single repair job, there are N repair_info object per shard.  The tracker object is the central place for the repair jobs.  repair_info objects are registered into the tracker object.

Reply:
repair_info depends on things like if the hint flush is successful or not. The hint flush process can take time.

Reply:
Well, having so much state scattered around is a recipe for confusion. Not to mention exception safety problems, updating multiple structures can get us in an inconsistent state on exception.

Reply:
repair_info seems very high-level. Is there a higher-level object?

Reply:
Yes, the repair tracker object is higher level which tracks individual per shard repair_info object.